src/Bonsai.ML.Torch/ToArray.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Xml.Serialization;
+using System.Linq.Expressions;
+using System.Reflection;
+using Bonsai.Expressions;
+using static TorchSharp.torch;
+
+namespace Bonsai.ML.Torch
+{
+    /// <summary>
+    /// Converts the input tensor into a flattened array of the specified element type.
+    /// </summary>
+    [Combinator]
+    [Description("Converts the input tensor into a flattened array of the specified element type.")]
+    [WorkflowElementCategory(ElementCategory.Transform)]
+    public class ToArray : SingleArgumentExpressionBuilder
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ToArray"/> class.
+        /// </summary>
+        public ToArray()
+        {
+            Type = typeof(double);
+        }
+
+        /// <summary>
+        /// Gets or sets the type mapping used to convert the input tensor into a flattened array.
+        /// </summary>
+        [Description("Gets or sets the type mapping used to convert the input tensor into a flattened array.")]
+        [TypeConverter(typeof(ScalarTypeConverter))]
+        [XmlIgnore]
+        public Type Type { get; set; }
```

Comment:
Would it be possible to use the `ScalarType` enum to pick a primitive data type here instead of a `Type`, similar to what happens with `ConvertScalarType`? Just wondering whether we could make it easier and more consistent to specify the type of the array here.